### PR TITLE
Removed C_* config definitions

### LINKED ISSUE #2819: Eliminate the Sidebar's Global Config Variables
This is a step towards Account- and Mailbox-specific config.

The Config System is [documented here](https://neomutt.org/dev/config).

At the end, it explains how to eliminate global config variables.
These two commits are where it has been applied to the Send library.

- [4e434fe556 send: get Config from a ConfigSubset](https://github.com/neomutt/neomutt/commit/4e434fe5564430ad23d00eb7886f8b5283eefd4c)
- [47a7ab79af send: no variable](https://github.com/neomutt/neomutt/commit/47a7ab79af5b3e2755c0e75dda168664bf835135)

This diagram is a little out-of-date, but it shows the layout of the Sidebar functions.
**Key**:
- **Blue club** - Sidebar config that is global
- **Red circle** - Sidebar config that will be Account-specific (soon)

<a href="https://flatcap.org/mutt/sidebar1.svg"></a>

## Sidebar Entry Points

The left-most circles are the entry points into the Sidebar.
In each of these, create a `ConfigSubset` and initialise it to the global config scope: `NeoMutt->sub`.

Each of the functions marked with a club or dot needs to be passed this `sub`.

## Config Elimination

1. Replace each reference to `C_SomeName` with a call to a config helper function
2. Remove the globals in `sidebar/config.c` entirely and update `config_init_sidebar()`.

### PR DESCRIPTION
**Pull request by paniash**
* **What does this PR do?***
Fixes #2819.

### CODE REVIEW

File: sidebar/window.c [hunk 1]
```diff
@@ -686,30 +693,34 @@ int sb_recalc(struct MuttWindow *win)
...
     // Otherwise, the full path will be indent, and it looks unusual.
-    if (C_SidebarFolderIndent && short_path_is_abbr)
+		const bool c_sidebar_folder_indent = cs_subset_bool(sub, "sidebar_folder_indent");
+    if (c_sidebar_folder_indent && short_path_is_abbr)
     {
-      if (C_SidebarComponentDepth > 0)
-        entry->depth -= C_SidebarComponentDepth;
+      const short c_sidebar_component_depth = cs_subset_number(sub, "sidebar_component_depth");
+			if (c_sidebar_component_depth > 0)
+				entry->depth -= c_sidebar_component_depth;
```
Comment (flatcap) [inline sidebar/window.c:721]: Spaces please, rather than tabs.
In vim:
``vim
set tabstop=2 shiftwidth=2 expandtab
```

**Review by flatcap — COMMENTED** [review]
Looking good so far.
You can use `NeoMutt->sub` (the global config scope) for all the `sub` parameters.

**Comment by paniash** [comment]
I haven't changed anything in `private.h` since I wasn't sure. Do I have to change the variables in it accordingly?

**Comment by flatcap** [comment]
> `private.h`

Yes, you need to remove the `extern C_Sidebar*`

Compiling the code will show you all the problems (there aren't too many).

I think there are a couple of `c_var`s that you haven't defined yet.

The remaining problems seem to be just typos.

The global is called `NeoMutt` not `Neomutt`

Lots of `const char const c_var = ` which make `c_var` a single `char`.
What's needed is a pointer `const char *c_var = `

**Comment by paniash** [comment]
> I think there are a couple of `c_var`s that you haven't defined yet.

I don't see any variables that I might've missed. Could you let me know?

**Review by flatcap — APPROVED** [review]

**Pull request merged**